### ISSUE TITLE
[Clarification] Why is the total batch size of 16 strictly required for training?

### ISSUE THREAD

**Issue by PaulokaxD** [tue-mps/eomt#72]
Hi folks,

I am trying to train EoMT from scratch using the provided instructions in the README. However, due to hardware constraints, I don't have access to 4 GPUs, and running a batch size of 4 per GPU on my current setup results in an Out of Memory (OOM) error.

The README explicitly states:

    ✅ Make sure the total batch size is devices × batch_size = 16

To avoid the OOM error, I experimented with different configurations where devices × batch_size <= 16. The code compiles and runs successfully without throwing errors. Since the script technically executes fine with a different total batch size, I'd like to understand the implications of this requirement. More concretely:

-  Will deviating from a total batch size of 16 severely hurt the model's convergence or final performance? Are there other important information that I may be missing by doing so?
- If I need to use a smaller total batch size, do you recommend scaling down the learning rate or modifying other hyperparameters to compensate?

Any insights or recommendations for single-GPU/low-resource training setups would be greatly appreciated! Thanks for the great work on this repository💕

**Comment by NiccoloCavagnero** [comment]
Hi @PaulokaxD, 

thanks for you interest in our work! 

We recommend using 16 as batch size as we tuned the other hyper-parameters accordingly. Since there are not Batch Norm operations in EoMT, you can safely enable gradient accumulation in the Trainer in order to match the desired batch size. This is what I personally did for most of the development phase.

However, you are not strictly required to use 16 and you can find a more suitable set of hyper-parameters for you setting. Usually, when changing batch size, you should scale at least the learning rate accordingly. Moreover, if you change the number of steps per epoch, you need to adjust the Mask Annealing schedule accordingly to match the new number of steps.

Best, 
Nick

**Comment by PaulokaxD** [comment]
Hi @NiccoloCavagnero,

thank you very much for the fast answer! Everything is more clear now:)

Bests,
Paula
